### ISSUE TITLE
PhantomJS discrepency

### ISSUE THREAD

**Issue by kasperlewau** [kasperlewau/angular-bind-notifier#16]
Ref. issues: #15 #14 #5 
---

There are some discrepencies between the following parameters; 

* PhantomJS 2.0 / 1.9.8
* Minified / non-minified version of Angular.
* `watchDelegate.length`, `/rx/.test(watchDelegate.toString())`, `watchDelegate.prototype`. 

--- 

What it boils down to, is that we're getting; 

a) False positives. 
b) Exponential registration of $on listeners. 
c) Broken bindings when B is "fixed". 

--- 

Will try to spend some more time on the issue this coming weekend, but right now I'm at a wee loss on what would actually cover the entire board.

**Comment by kasperlewau** [comment]
Output of `log(watchDelegate.length, watchDelegate.toString)`

```
# phantomjs2
4, 'function wrap() {[native code]}'

# phantomjs 1.9.8
0, 'function () {
  return fToBind.apply(this instanceof fNOP
  ? this
  : oThis,
  aArgs.concat(Array.prototype.slice.call(arguments)));
}'
```

There's the reason for false positivies on differing versions of Phantom. I'm leaning towards scrapping that and just running Chrome/FF launchers instead.

**Comment by kasperlewau** [comment]
Or rather, the *need* for a `.bind` polyfill.

**Comment by kasperlewau** [comment]
`if (watchDelegate.prototype && Object.getOwnPropertyNames(watchDelegate.prototype).indexOf('constructor') > -1)` 

appears to have taken the cake. will close this down once I've verified the behaviour in ze browser.

**Comment by kasperlewau** [comment]
Closed by fabf9ea6308ccbe6519f76c0a5e93fcc65e79bda
